components/script/script_thread.rs

```diff
@@ -173,7 +174,7 @@ pub(crate) fn with_script_thread<R: Default>(f: impl FnOnce(&ScriptThread) -> R)
 pub(crate) unsafe fn trace_thread(tr: *mut JSTracer) {
     with_script_thread(|script_thread| {
         trace!("tracing fields of ScriptThread");
-        script_thread.trace(tr);
+        unsafe { script_thread.trace(tr) };
```

Comment:
Why is this change necessary?

Reply:
Saw an warning `call to unsafe function is unsafe and requires an unsafe function or block` so I applied the suggestion from rust analyzer, I can remove it if that's not desired

Reply:
I think you must be using a different version of rustc than Servo or something. Let's leave this change out of the PR.

Reply:
Hmm that's weird, I'm using `rustc 1.85.0 (4d91de4e4 2025-02-17)`, the same as the setting in `rust-toolchain.toml`, anyways, removed